Add Fleet Provisioning to README (#1682)
aws/aws-iot-device-sdk-embedded-C: README.md

See memory requirements for the latest release [here](https://docs.aws.amazon.com/embedded-csdk/202103.00/lib-ref/libraries/aws/ota-for-aws-iot-embedded-sdk/docs/doxygen/output/html/index.html#ota_memory_requirements).

#### AWS IoT Fleet Provisioning

> The [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisoning-for-aws-iot-embedded-sdk) library enables you to interact with the AWS IoT Fleet Provisioning MQTT APIs in order to provison IoT devices that do not have device certificates. With AWS IoT Fleet Provisioning, devices can securely recieve unique device certificates from AWS IoT when they connect for the first time. For an overview of all provisioning options offered by AWS IoT, see [Device provisioning](https://docs.aws.amazon.com/iot/latest/developerguide/iot-provision.html). For details about Fleet Provisioning, refer to the [AWS IoT Fleet Provisioning documentation](https://docs.aws.amazon.com/iot/latest/developerguide/provision-wo-cert.html).

Reviewer: ```suggestion
The [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisoning-for-aws-iot-embedded-sdk) library enables you to interact with the [AWS IoT Fleet Provisioning MQTT APIs](https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html) in order to provison IoT devices that do not have device certificates. With AWS IoT Fleet Provisioning, devices can securely recieve unique device certificates from AWS IoT when they connect for the first time. For an overview of all provisioning options offered by AWS IoT, see [Device provisioning](https://docs.aws.amazon.com/iot/latest/developerguide/iot-provision.html). For details about Fleet Provisioning, refer to the [AWS IoT Fleet Provisioning documentation](https://docs.aws.amazon.com/iot/latest/developerguide/provision-wo-cert.html).
```

> The [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisoning-for-aws-iot-embedded-sdk) library enables you to interact with the AWS IoT Fleet Provisioning MQTT APIs in order to provison IoT devices that do not have device certificates. With AWS IoT Fleet Provisioning, devices can securely recieve unique device certificates from AWS IoT when they connect for the first time. For an overview of provisioning options offered by AWS IoT, see [Device provisioning](https://docs.aws.amazon.com/iot/latest/developerguide/iot-provision.html). For details about Fleet Provisioning can be found in the [AWS IoT Fleet Provisioning documentation](https://docs.aws.amazon.com/iot/latest/developerguide/provision-wo-cert.html).

Reviewer: ```suggestion
The [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisoning-for-aws-iot-embedded-sdk) library enables you to interact with the AWS IoT Fleet Provisioning MQTT APIs in order to provison IoT devices that do not have device certificates. With AWS IoT Fleet Provisioning, devices can securely recieve unique device certificates from AWS IoT when they connect for the first time. For an overview of all provisioning options offered by AWS IoT, see [Device provisioning](https://docs.aws.amazon.com/iot/latest/developerguide/iot-provision.html). For details about Fleet Provisioning, refer to the [AWS IoT Fleet Provisioning documentation](https://docs.aws.amazon.com/iot/latest/developerguide/provision-wo-cert.html).
```

> The [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisioning-for-aws-iot-embedded-sdk) library enables you to interact with the [AWS IoT Fleet Provisioning MQTT APIs](https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html) in order to provison IoT devices that do not have device certificates. With AWS IoT Fleet Provisioning, devices can securely recieve unique device certificates from AWS IoT when they connect for the first time. For an overview of all provisioning options offered by AWS IoT, see [Device provisioning](https://docs.aws.amazon.com/iot/latest/developerguide/iot-provision.html). For details about Fleet Provisioning, refer to the [AWS IoT Fleet Provisioning documentation](https://docs.aws.amazon.com/iot/latest/developerguide/provision-wo-cert.html).

Reviewer 2: ```suggestion
The [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisioning-for-aws-iot-embedded-sdk) library enables you to interact with the [AWS IoT Fleet Provisioning MQTT APIs](https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html) in order to provison IoT devices without preexisting device certificates. With AWS IoT Fleet Provisioning, devices can securely receive unique device certificates from AWS IoT when they connect for the first time. For an overview of all provisioning options offered by AWS IoT, see [device provisioning documentation](https://docs.aws.amazon.com/iot/latest/developerguide/iot-provision.html). For details about Fleet Provisioning, refer to the [AWS IoT Fleet Provisioning documentation](https://docs.aws.amazon.com/iot/latest/developerguide/provision-wo-cert.html).
```
